tests/functional/test_install.py

```diff
@@ -1821,3 +1821,53 @@ def test_install_sends_client_cert(install_args, script, cert_factory, data):
         environ, _ = call_args.args
         assert "SSL_CLIENT_CERT" in environ
         assert environ["SSL_CLIENT_CERT"]
+
+
+def test_install_skip_work_dir_pkg(script, data):
+    """
+    Test that install of a package in working directory
+    should pass on the second attempt after an install
+    and an uninstall
+    """
+
+    # Create a test package and install it
+    pkg_path = create_test_package_with_setup(
+        script, name='simple', version='1.0')
+    script.pip('install', '-e', '.',
+               expect_stderr=True, cwd=pkg_path)
+
+    # Uninstalling the package and installing it again will succeed
+    script.pip('uninstall', 'simple', '-y')
+
+    result = script.pip('install', '--find-links',
+                        data.find_links, 'simple',
+                        expect_stderr=True, cwd=pkg_path)
+
+    assert 'Requirement already satisfied: simple' not in result.stdout
+    assert 'Successfully installed simple' in result.stdout
+
+
+def test_install_include_work_dir_pkg(script, data):
+    """
+    Test that install of a package in working directory
+    should fail on the second attempt after an install
+    if working directory is added in PYTHONPATH
+    """
+
+    # Create a test package and install it
+    pkg_path = create_test_package_with_setup(
+        script, name='simple', version='1.0')
+    script.pip('install', '-e', '.',
+               expect_stderr=True, cwd=pkg_path)
+
+    # Uninstall will fail with given warning
```

Comment:
I have reworded the comment accordingly. Please take a look

Reply:
This comment still seems misleading to me :thinking: 
The `script.pip('uninstall', 'simple', '-y')` does not seem to fail or print warnings since there is no `expect_stderr=True`

Reply:
Created https://github.com/pypa/pip/pull/8012 to address this comment.